openedx/features/qverse_features/registration/models.py

```diff
@@ -0,0 +1,96 @@
+"""
+Django models for qverse registration app.
+"""
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.utils.translation import ugettext_noop
+
+from .validators import validate_file_extension, validate_current_level, validate_programme_choices
+
+
+# explicitly saving lengths of these generators
+# as these would be used in validtors.py for
+# validations. As these are the generators they would
+# be exhausted after single use, so we couldn't get their
+# length in validators.py
+MAX_PROGRAMME_CHOICES = 6
+PROGRAMME_CHOICES = (
+    (1, ugettext_noop('Undergraduate Studies')),
+    (2, ugettext_noop('Postgraduate Studies')),
+    (3, ugettext_noop('Matured Student Programme (MSP)')),
+    (4, ugettext_noop('Diploma')),
+    (5, ugettext_noop('Pre-Science')),
+    (6, ugettext_noop('Sandwish')),
+)
+MAX_LEVEL_CHOICES = 8
+LEVEL_CHOICES = ((i, ugettext_noop('Year ' + str(i))) for i in range(1, MAX_LEVEL_CHOICES))
+
+
+class BulkUserRegistration(models.Model):
+    """
+    Saves csv records for bulk user registration.
+    """
+    record = models.FileField(
+        upload_to='qverse_registration_csvs/',
+        blank=False,
+        null=True,
+        validators = [validate_file_extension]
+    )
+    description = models.CharField(max_length=100, blank=False, null=True)
+
+    class Meta(object):
+        app_label = 'registration'
+
+    def __str__(self):
+        return self.description
+
+
+class Department(models.Model):
+    """
+    Represents educational departments.
+    """
+    department_number = models.IntegerField(unique=True, validators=[MinValueValidator(1)])
+    department_name = models.CharField(unique=True, max_length=50)
+
+    def save(self, *args, **kwargs):
+        """
+        Make sure that all the department names are saved in db in title case.
+
+        This would help us to ensure unique constraint.
+        """
+        self.department_name = self.department_name.title()
+        return super(Department, self).save(*args, **kwargs)
+
+    def __str__(self):
+        return self.department_name
+
+
+
+class UserProfile(models.Model):
```

Comment:
please make sure, I think we should rename this model

Reply:
Any suggestions. I came up with this after considering a lot of other options.
If you think that it would mix with the `UserProfile` model of edx, it wouldn't because I have given separate `related_name` for this in `user` field.

Reply:
`QVerseUserProfile`, `ExtendedUserProfile`

we should not depend on `related_name`, because it isn't always required value.

Reply:
How about UserExtendedProfile?

Reply:
Picked `QVerseUserProfile`